packages/dashboard/src/managers/resource-manager.ts

```diff
@@ -71,6 +80,12 @@ export default class ResourceManager {
     this.attributionPrefix = resources.attributionPrefix || 'OSRC-SG';
     this.cartIds = resources.cartIds || [];
     this.loggedInDisplayLevel = resources.loggedInDisplayLevel;
+    this.supportedTasks = resources.supportedTasks || [
+      DefaultPatrolTaskDefinition,
+      DefaultDeliveryTaskDefinition,
+      DefaultComposeCleanTaskDefinition,
+      DefaultCustomComposeTaskDefinition,
+    ];
```

Comment:
iirc, we cannot support a fully new task definitoin yet, we only support task definition ids which is hardcoded in create task form. Either validate that the `supportedTasks` have a valid task definition ids or make the TaskDefinition typing more strict, e.g. `taskDefinitiodId: 'foo' | 'bar'`.

Reply:
the resource manager side will do the checking and only allow supported tasks,
https://github.com/open-rmf/rmf-web/pull/925/commits/7f7eb6203121b1f1631da5978761d2b86a2eb4ec